File: Problems 001-050/p06.hs
sqos :: Integer -> Integer
sqos n = (^2) $ n * (n + 1) `div` 2

sosq :: Integer -> Integer
sosq n = n * (n + 1) * (2 * n + 1) `div` 6

solve = sequence [sqos, negate . sosq]

main :: IO()
main = print $ sum . solve $ 100

File: Problems 001-050/p06.py
def sosq(n: int) -> int:
    assert n > 0
    return n * (n + 1) * (2 * n + 1) // 6


def sqos(n: int) -> int:
    assert n > 0
    return (n * (n + 1) // 2) ** 2


print(sqos(100) - sosq(100))
